### PR TITLE
Updates Project

### PR DESCRIPTION
**Pull request by kikejimenez**
In this version of the project:  
  - The train-test split is balanced (sentiment ratio is conserved in each split).
  - To assure reproducibility shuffles are only done inside Tensorflow
  - The `__del__`  destructor in  `Pipeline`,  has been customized to erase the `estimator` directory
  - Test have been incorporated to compare different instances of `Pipeline` for equal initial conditions
  - The K fold validation has been updated with only pandas routines (no sklearn)
  - The K fold keeps track of the best score through the random seed
  - The index now is *reproducible/deterministic*

### CODE REVIEW

**Comment by kikejimenez** [comment]
The CI is not yet integrated and these checks are ommited
